diesel/src/query_builder/ast_pass.rs

```diff
@@ -3,8 +3,21 @@ use query_builder::{BindCollector, QueryBuilder};
 use result::QueryResult;
 use types::{ToSql, HasSqlType};
 
-#[doc(hidden)]
 #[allow(missing_debug_implementations)]
+/// The primary type used when walking a Diesel AST during query execution
+///
+/// Executing a query is generally done in multiple passes. This list includes,
+/// but is not limited to:
+///
+/// - Generating the SQL
+/// - Collecting and serializing the values sent separately from the SQL
```

Comment:
This sentence is weird IMO.

What about "Collecting and serializing SQL bound values" ?